thoth/adviser/wraps/intel_tensorflow.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2020 Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A wrap that notifies about Intel TensorFlow builds."""
+
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import TYPE_CHECKING
+
+from ..state import State
+from ..wrap import Wrap
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+
+class IntelTensorFlowWrap(Wrap):
+    """A wrap that recommends using Intel TensorFlow if TensorFlow is in resolved dependencies.
+
+    https://software.intel.com/content/www/us/en/develop/articles/intel-optimization-for-tensorflow-installation-guide.html#pip_wheels
+    """
+
+    # Sandy bridge CPUID taken from https://en.wikipedia.org/wiki/Sandy_Bridge
+    # Ivy bridge CPUID taken from https://en.wikipedia.org/wiki/Ivy_Bridge_(microarchitecture)
+    #
+    # As CPUID encodes family (bits 8 - 11 with mask 0xF00) and model (bits 4 - 7 with mask 0xF0) we got these
+    # numbers from CPUID.
+    _CPU_TABLE = frozenset(
+        {
+            # tuple (model, family)
+            # Sandy Bridge-HE-4, Sandy Bridge-H-2, Sandy Bridge-M-2
+            # ((0x0206A7 & 0xF0) >> 4, (0x0206A7 & 0xF00) >> 8),
+            # Sandy Bridge - EP - 8
+            # ((0x0206D6 & 0xF0) >> 4, (0x0206D6 & 0xF00) >> 8),
+            # ((0x0206D7 & 0xF0) >> 4, (0x0206D7 & 0xF00) >> 8),
+            # Sandy Bridge - EP - 4
+            # ((0x0206D6 & 0xF0) >> 4, (0x0206D6 & 0xF00) >> 8),
+            # ((0x0206D7 & 0xF0) >> 4, (0x0206D7 & 0xF00) >> 8),
+            # Ivy Bridge-M-2, Ivy Bridge-H-2, Ivy Bridge-HM-4, Ivy Bridge-HE-4
+            # ((0x000306A9 & 0xF0) >> 4, (0x000306A9 & 0xF00) >> 8),
+            # All maps to the following values:
+            (13, 6),
```

Comment:
I think we should include newer intel-chips as well. Though the binaries won't be the absolute best they will at least use `intel` specific instruction set which should impart some level of performance increase.

Reply:
Definitely +1. I've added a note to https://github.com/thoth-station/adviser/issues/1022. I'm not an expert on CPU models/families, so we will need to review this. The listing for our AVX builds is on the list as well. I did not find any reliable enough sources for these and did not want to spend much time on the initial implementation that shows the idea.

Reply:
Then this looks good to me. Thanks for opening an issue.